libminifi/include/FlowController.h

```diff
@@ -234,7 +233,11 @@ class FlowController : public core::controller::ForwardingControllerServiceProvi
   std::chrono::steady_clock::time_point start_time_;
 
  private:
-  void getProcessorController(const std::string& name, std::vector<state::StateController*>& controllerVec,
+  std::vector<state::StateController*> getAllComponents();
+
+  state::StateController* getComponent(const std::string &name);
+
+  void getProcessorController(const std::string& name, state::StateController*& controller,
                               const std::function<std::unique_ptr<state::ProcessorController>(core::Processor&)>& controllerFactory);
```

Comment:
this could return the `controller` instead of using the slightly strange-looking `state::StateController*&` out parameter:
```suggestion
  state::StateController* getProcessorController(const std::string& name,
                              const std::function<std::unique_ptr<state::ProcessorController>(core::Processor&)>& controllerFactory);
```

Reply:
Done.